application/src/main/java/org/opentripplanner/transit/model/filter/transit/RegularStopMatcherFactory.java

```diff
@@ -0,0 +1,50 @@
+package org.opentripplanner.transit.model.filter.transit;
+
+import java.util.function.Predicate;
+import org.opentripplanner.transit.api.request.RegularStopRequest;
+import org.opentripplanner.transit.model.filter.expr.EqualityMatcher;
+import org.opentripplanner.transit.model.filter.expr.ExpressionBuilder;
+import org.opentripplanner.transit.model.filter.expr.GenericUnaryMatcher;
+import org.opentripplanner.transit.model.filter.expr.Matcher;
+import org.opentripplanner.transit.model.site.RegularStop;
+
+/**
+ * A factory for creating matchers for {@link RegularStop} objects.
+ * <p/>
+ * This factory is used to create matchers for {@link RegularStop} objects based on a request. The
+ * resulting matcher can be used to filter a list of {@link RegularStop} objects.
+ */
+public class RegularStopMatcherFactory {
+
+  /**
+   * Creates a matcher that filters {@link RegularStop} objects with the provided {@link RegularStopRequest}
+   * and {@link Predicate}. The {@link Predicate} is used to determine if a {@link RegularStop} is
+   * in use. Typically the inUseProvider is a function that checks if the {@link RegularStop} is in
+   * a set of used stops.
+   * @param request - {@link RegularStopRequest} to filter {@link RegularStop} objects.
+   * @param inUseProvider - {@link Predicate} to determine if a {@link RegularStop} is in use.
+   * @return - {@link Matcher} for {@link RegularStop} objects.
+   */
+  public static Matcher<RegularStop> of(
+    RegularStopRequest request,
+    Predicate<RegularStop> inUseProvider
+  ) {
+    ExpressionBuilder<RegularStop> expr = ExpressionBuilder.of();
+
+    if (request.agency() != null) {
+      expr.matches(agency(request.agency()));
+    }
+    if (request.filterByInUse()) {
+      expr.matches(inUseMatcher(inUseProvider));
+    }
+    return expr.build();
+  }
+
+  static Matcher<RegularStop> agency(String agency) {
+    return new EqualityMatcher<>("agency", agency, stop -> stop.getId().getFeedId());
```

Comment:
It appears that you're copying the old logic which compares the argument "authority" with the feed id of the the stop but in Entur's case all stops have the same feed id. This is pretty supicious. Is this really what you want to do?

Reply:
Also, stops don't "have" a single agency. Stops have 0 or more routes which in turn have an agency so there can be multiple.

I think you want to go back and figure out what the requirements really are.

Reply:
This probably left-over from the days this was implemented at HSL. I am pretty sure it is not the agency, but the feedId. We should not assume here that all stops comes from NSR (Norway National Stop Register).

So, the API should not be changed, but all internal OTP classes should refer to this as `feedId`.